apps/dashboard/app/helpers/batch_connect/sessions_helper.rb

```diff
@@ -95,7 +95,7 @@ def time(session)
   end
 
   def host(session)
-    if ::Configuration.ood_bc_ssh_to_compute_node
+    if ::Configuration.ood_bc_ssh_to_compute_node && session.cluster.batch_connect_ssh_allow?
```

Comment:
I think we need to put all this logic into a function, and have precedence like we do below. So you could set it globally to true, then false for one cluster (or vice versa).  The more specific the setting, the higher the precedence. So cluster's always override the global setting.

https://github.com/OSC/ondemand/blob/09cff737c438118c2d89fafef03312bb72e66331/apps/dashboard/app/models/batch_connect/session_context.rb#L86-L98

Reply:
I pushed a new commit with something similar to what you linked, just seems much more verbose than necessary since everything we are dealing with are boolean functions so feels like this might be sufficient:

```ruby
def ssh_to_compute_node?
  cluster_ssh_to_compute_node? && global_ssh_to_compute_node?
ebd
```

Reply:
So the simpler cases can be handled by this:

```ruby
def ssh_to_compute_node?
  cluster_ssh_to_compute_node? && global_ssh_to_compute_node?
end
```

If any of those is false, the SSH to node is disabled so you just modify 1 thing to turn it off depending on how you want to turn it off.  So if you disable it globally , the default `true` for cluster is not going to matter. Same for disabling at the cluster level, if you disable at cluster and global is true, then the SSH is still getting disabled because not all evaluate to true.  This only takes care of making it easy to opt-out of individual clusters since the current behavior is to enable always.  To disable globally and enable for 1 cluster might require changes to ood_core so it doesn't return a boolean but can return nil.  Does that sound correct?  So `batch_connect_ssh_allow?` from ood_core will only return the boolean defined in cluster YAML and if no value is defined it returns `nil` and then we can take that and make decisions in the dashboard.

Reply:
| To disable globally and enable for 1 cluster might require changes to ood_core so it doesn't return a boolean but can return nil. Does that sound correct?

Reply:
the `ood_core` we currently use returns nil as the default now, so we're ready to move forward with that.

Reply:
Updated the logic to handle nil